template/android/settings.gradle

```
@@ -1,11 +1,22 @@
 rootProject.name = 'HelloWorld'
+
 apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
 include ':app'
 includeBuild('../node_modules/react-native-gradle-plugin') {
     dependencySubstitution {
         substitute(module("com.facebook.react:react")).using(project(":"))
     }
 }
+includeBuild("../node_modules/react-native/packages/annotations-compiler/annotations") {
+    dependencySubstitution {
+        substitute(module("com.facebook.react:annotations")).using(project(":"))
+    }
+}
+includeBuild("../node_modules/react-native/packages/annotations-compiler/compiler") {
+    dependencySubstitution {
+        substitute(module("com.facebook.react:annotations-compiler")).using(project(":"))
+    }
+}
```

```suggestion
includeBuild("../node_modules/react-native/packages/annotations-compiler/annotations")
includeBuild("../node_modules/react-native/packages/annotations-compiler/compiler")
```

This should be enough as long as you:
1. Add a `group = "com.facebook.react"` in both `.gradle` files.
2. Apply the dependency as `annotationProcessor("com.facebook.react:annotations")` and `annotationProcessor("com.facebook.react:compiler")`. Maybe rename the project folder as just `compiler` is not informative enough.